src/main/java/com/ParqueCore/ParkBeto/service/impl/AtracaoService.java

```diff
@@ -30,13 +31,37 @@ public Atracao createAtracao(Atracao atracao) {
         }
     }
 
+
+    public List<Atracao> listaAtracoes(){
+        return atracaoRepository.findAll();
+    }
+
+    public Atracao buscarPorId(long id){
+        return atracaoRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Atracao nao encontrada"));
+    }
+
+    public Atracao buscarPorNome(String nome){
+        return atracaoRepository.findByNome(nome)
+                .orElseThrow(() -> new EntityNotFoundException("Atracao com o nome: '"+ nome +"' nao foi encontrada"));
+    }
+    public List<Atracao> buscarPorTipo(AtracaoTipo tipo){
```

Comment:
No método buscarPorTipo, se nenhum resultado for encontrado, você pode lançar uma exceção ou retornar uma lista vazia com um código de status adequado, como 204 No Content na controller.